craft_application/models/project.py

```diff
@@ -67,3 +78,9 @@ def effective_base(self) -> Any:  # noqa: ANN401 app specific classes can improv
         if self.base is not None:
             return self.base
         raise RuntimeError("Could not determine effective base")
+
+    def get_build_plan(self) -> List[BuildInfo]:
```

Comment:
question about this method: is it supposed to return *all* combinations from the project file, or only those that match the running host? 
For example, if a project has something like:
```
architectures:
  - build-on: amd64
    build-for: amd64
  - build-on: [amd64, arm64]
    build-for: arm64
```

... and the application is running on `arm64` should this method return a list with a) 3 build infos, or b) only the 1 build info that can run? If the options is "a", where does the code then "filter out" the invalid build-ons?

Reply:
Following the convention used in Snapcraft it should return only the builds it can run.

Reply:
If this is the case, then I think we can make the lives of the applications easier:
- have a `get_full_build_plan()` (or some other name) that applications must implement to return the full, unfiltered build list from the project file
- have this `get_build_plan()` method be implemented by this class to call `get_full_build_plan()` and then do the filtering on the result based on valid `build-ons`.
Otherwise we're just going to have a lot of almost-identical code in the applications

Reply:
We can set `get_build_plan()` to return the full list, and have code in Craft-Application to filter it according to the host platform. Wdyt?

Reply:
That's probably fine but [this code](https://github.com/canonical/craft-application/pull/70/files#diff-c395b65ed7321f1bbf3c026ade9775f6a47275ae4051f4981b0376c1730c25d4R162) will have to be updated to use that filter. That's one reason why I don't like that approach too much, it becomes kind of easy to "forget" to filter and iterate on the full build plan, which most times is not what you want to do right?

Reply:
We're also filtering by build-for there, and applications generally won't override `run_managed` -- is Rockcraft doing that?

Reply:
Rockcraft is not doing that, but you also have to filter by the `build-on` that matches the host arch, no?

Reply:
Changed the current `build-for` filter to also filter for `build-on`. The application-specific implementation of `get_build_plan()` now should return all possibilities. This may require further refactoring when named platforms are supported (so one could specify `--build-for <platform name>`.